question28/js/forgotpassword.js

```diff
@@ -19,21 +19,44 @@ mailInputArea.addEventListener('blur', (e) => {
   toggleDisabledOfSubmitButton(checkAllValidity());
 });
 
-const init = () => {
-  const inputMailData = mailInputArea.value;
+const checkData = (emailAddressValue) => {
   const registeredUserData = JSON.parse(localStorage.getItem('userData'));
+  return ((emailAddressValue === registeredUserData.user_email));
+}
 
-  if (registeredUserData === null) {
-    window.location.href = './notautherize.html';
-    return; 
+const userDataVerification = () => {
+  const userData = mailInputArea.value;
+  return new Promise((resolve,reject) => {
+    if (checkData(userData)){
+      resolve({ token: chance.apple_token(), ok: true, code: 200 });
+    } else {
+      reject({ ok: false, code: 401 }); 
+    }
+  })
+}
+
+const checkExistenceOfUserdata = async() => {
+  try {
+    return await userDataVerification();
+  }
+  catch (e) {
+    console.error(e);
   }
-  if (registeredUserData.user_email === inputMailData) {
-    const tokenForPasswordReset = chance.apple_token();
-    const passwordResetPageUrl = "./register/password.html";
-    window.location.href = `${passwordResetPageUrl}?token=${tokenForPasswordReset}`;
+}
+
+const init = async () => {
+  const verificationResult = await checkExistenceOfUserdata();
+
+  if (!verificationResult) {
+    mailInputArea.nextElementSibling.textContent = "会員登録されていないか入力値が間違っています";
+    submitButton.disabled = true;
     return; 
   }
-  window.location.href = './notautherize.html';
+
+  const tokenForPasswordReset = verificationResult.token;
+  const passwordResetPageUrl = "./register/password.html";
+  localStorage.setItem('tokenForforgotPassword', tokenForPasswordReset);
+  window.location.href = `${passwordResetPageUrl}?token=${tokenForPasswordReset}`;
```

Comment:
②forgotpassword.js
Emits an error if the value in the local storage does not match the input value. 
If the value in local storage matches the input value, a token is embedded and the user is moved to password.html with the token as a parameter.